src/ports/trending/component.ts

```diff
@@ -0,0 +1,103 @@
+import seedrandom from 'seedrandom'
+import { Item, Sale, SaleFilters, SaleSortBy } from '@dcl/schemas'
+
+import { IMergerComponent } from '../merger/types'
+import { getDateXDaysAgo } from '../analyticsDayData/utils'
+import { IItemsComponent } from '../items/types'
+import { ITrendingsComponent, TrendingFilters } from './types'
+
+const DEFAULT_SIZE = 20
+const SALES_CUT = 0.6
+const VOLUME_CUT = 0.4
+
+export function createTrendingsComponent(
+  salesComponent: IMergerComponent<Sale, SaleFilters, SaleSortBy>,
+  itemsComponent: IItemsComponent
+): ITrendingsComponent {
+  /**
+   * The fetch will return the trending NFTs based on the sales amount and volume.
+   * The current logic gets the 60% with more sales and the 40% with more traded volume. Then uses seedrandom to shuffle the concatenated array in a deterministic order.
+   * @param filters TrendingFilters
+   * @returns NFT
+   */
+  async function fetch(filters: TrendingFilters) {
+    // Fetch all sales from the past 24hs
+    const sales = await salesComponent.fetch({
+      from: getDateXDaysAgo(1).getTime(),
+      sortBy: SaleSortBy.MOST_EXPENSIVE, // sort by volume
+    })
+
+    // Fetch all the items from those 24hs sales
+    const items = (
+      await Promise.all(
+        sales.map((sale) =>
+          itemsComponent.fetch({
+            contractAddress: sale.contractAddress,
+            itemId: sale.itemId || undefined,
+          })
+        )
+      )
+    ).reduce((a, b) => a.concat(b), []) // flatten the array of arrays
+
+    // Get trending sales by amount of sales
+    const trendingSales = sales.reduce((acc, sale) => {
+      const key = `${sale.contractAddress}-${sale.itemId}`
```

Comment:
Watch out here: the `sale.itemId` could be `undefined` for NFTs that are not part of a collection (parcels, estates, names, and L1 wearables). Those should be skipped from the `trendingSales` record, ie:

```ts
const trendingSales = sales.reduce((acc, sale) => {
  if (sale.itemId) {
    const key = `${sale.contractAddress}-${sale.itemId}`
    acc[key] = (acc[key] || 0) + 1
  }
  return acc
}, {} as Record<string, number>)
```